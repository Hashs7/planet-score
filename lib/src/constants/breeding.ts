/**
 *
 * @type {{MEDIUM: string, BAD: string, GOOD: string}}
 */
export enum BREEDING_CONDITION {
  GOOD = 'v',
  MEDIUM = 'o',
  BAD = 'r'
}

/**
 * Color state for breeding score
 *
 * @type {{[p: string]: string}}
 */
export const BREEDING_CONDITION_COLOR: { [index: string]: string } = {
  [BREEDING_CONDITION.GOOD]: '#007C3B',
  [BREEDING_CONDITION.MEDIUM]: '#F07E02',
  [BREEDING_CONDITION.BAD]: '#E33116'
};

/**
 * List of available animals
 *
 * @type {{OTHER: string, BEE: string, FARMING_FISH: string, FISHING_FISH: string, CHICKEN: string, SHEEP: string, DUCK: string, COW: string, GOAT: string, RABBIT: string, PIG: string}}
 */
export enum BREEDING_ANIMAL {
  COW = 'bovin',
  CHICKEN = 'poule',
  SHEEP = 'brebis',
  GOAT = 'chevre',
  PIG = 'porc'

  // Icon not integrated
  // BEE = 'abeille',
  // DUCK = 'canard',
  // RABBIT = 'lapin',
  // FARMING_FISH = 'poisson_aquaculture',
  // FISHING_FISH = 'poisson_peche',
  // OTHER = 'autre'
}

/**
 * Regex to extract only breeding animal word
 *
 * @type {RegExp}
 */
export const BREEDING_ANIMAL_REGEX = new RegExp(`(${Object.values(BREEDING_ANIMAL).join('|')})`);
