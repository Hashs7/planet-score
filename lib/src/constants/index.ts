export * from './breeding';
export * from './score';
