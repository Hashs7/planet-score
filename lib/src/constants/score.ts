/**
 * The score values.
 */
export enum ScoreValues {
  A = 'A',
  B = 'B',
  C = 'C',
  D = 'D',
  E = 'E'
}

/**
 * The score values in order of best to worst.
 */
export const SCORE_POSITIONS = [
  ScoreValues.A,
  ScoreValues.B,
  ScoreValues.C,
  ScoreValues.D,
  ScoreValues.E
];
