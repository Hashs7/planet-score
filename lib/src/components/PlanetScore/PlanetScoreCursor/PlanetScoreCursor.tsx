import { SCORE_POSITIONS, ScoreValues } from '@/constants';
import { FunctionComponent } from 'react';

interface PlanetScoreCursorProps {
  score: ScoreValues;
  translateY?: number;
}

const PlanetScoreCursor: FunctionComponent<PlanetScoreCursorProps> = ({
  score = '',
  translateY = 0
}) => {
  const index = SCORE_POSITIONS.indexOf(score.toUpperCase() as ScoreValues);
  const translateX = index * 108;

  if (index === -1) {
    return null;
  }

  return (
    <g className="cursor" transform={`translate(${translateX} ${translateY})`}>
      <path
        style={{ fill: '#FDFDFD' }}
        d="M572.2,346.6c-16,0-29,13-29,29s13,29,29,29c16,0,29-13,29-29S588.1,346.6,572.2,346.6z"
      />
      <path
        style={{ fill: '#273259' }}
        d="M572.2,336.6c-21.5,0-39,17.5-39,39s17.5,39,39,39c21.5,0,39-17.5,39-39S593.7,336.6,572.2,336.6z
		 M543.2,375.6c0-16,13-29,29-29c16,0,29,13,29,29s-13,29-29,29C556.2,404.6,543.2,391.6,543.2,375.6z"
      />
    </g>
  );
};

export default PlanetScoreCursor;
