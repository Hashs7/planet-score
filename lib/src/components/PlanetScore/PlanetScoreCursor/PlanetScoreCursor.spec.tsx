import { render } from '@testing-library/react';
import PlanetScoreCursor from './PlanetScoreCursor';
import { SCORE_POSITIONS, ScoreValues } from '@/constants';
import { SvgWrapper } from '@tests/utils';

describe('<PlanetScoreCursor>', () => {
  it('should set translateY value', () => {
    const { container } = render(<PlanetScoreCursor score={ScoreValues.A} translateY={200} />, {
      wrapper: SvgWrapper
    });

    expect(container).toMatchSnapshot();
  });

  SCORE_POSITIONS.forEach((score) => {
    it(`(${score}) renders the PlanetScoreCursor`, () => {
      const { container } = render(<PlanetScoreCursor score={score} />, { wrapper: SvgWrapper });

      expect(container).toMatchSnapshot();
    });
  });
});
