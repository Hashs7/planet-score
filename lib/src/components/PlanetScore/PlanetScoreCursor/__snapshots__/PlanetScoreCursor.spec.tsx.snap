// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlanetScoreCursor> (A) renders the PlanetScoreCursor 1`] = `
<div>
  <svg>
    <g
      class="cursor"
      transform="translate(0 0)"
    >
      <path
        d="M572.2,346.6c-16,0-29,13-29,29s13,29,29,29c16,0,29-13,29-29S588.1,346.6,572.2,346.6z"
        style="fill: #FDFDFD;"
      />
      <path
        d="M572.2,336.6c-21.5,0-39,17.5-39,39s17.5,39,39,39c21.5,0,39-17.5,39-39S593.7,336.6,572.2,336.6z
		 M543.2,375.6c0-16,13-29,29-29c16,0,29,13,29,29s-13,29-29,29C556.2,404.6,543.2,391.6,543.2,375.6z"
        style="fill: #273259;"
      />
    </g>
  </svg>
</div>
`;

exports[`<PlanetScoreCursor> (B) renders the PlanetScoreCursor 1`] = `
<div>
  <svg>
    <g
      class="cursor"
      transform="translate(108 0)"
    >
      <path
        d="M572.2,346.6c-16,0-29,13-29,29s13,29,29,29c16,0,29-13,29-29S588.1,346.6,572.2,346.6z"
        style="fill: #FDFDFD;"
      />
      <path
        d="M572.2,336.6c-21.5,0-39,17.5-39,39s17.5,39,39,39c21.5,0,39-17.5,39-39S593.7,336.6,572.2,336.6z
		 M543.2,375.6c0-16,13-29,29-29c16,0,29,13,29,29s-13,29-29,29C556.2,404.6,543.2,391.6,543.2,375.6z"
        style="fill: #273259;"
      />
    </g>
  </svg>
</div>
`;

exports[`<PlanetScoreCursor> (C) renders the PlanetScoreCursor 1`] = `
<div>
  <svg>
    <g
      class="cursor"
      transform="translate(216 0)"
    >
      <path
        d="M572.2,346.6c-16,0-29,13-29,29s13,29,29,29c16,0,29-13,29-29S588.1,346.6,572.2,346.6z"
        style="fill: #FDFDFD;"
      />
      <path
        d="M572.2,336.6c-21.5,0-39,17.5-39,39s17.5,39,39,39c21.5,0,39-17.5,39-39S593.7,336.6,572.2,336.6z
		 M543.2,375.6c0-16,13-29,29-29c16,0,29,13,29,29s-13,29-29,29C556.2,404.6,543.2,391.6,543.2,375.6z"
        style="fill: #273259;"
      />
    </g>
  </svg>
</div>
`;

exports[`<PlanetScoreCursor> (D) renders the PlanetScoreCursor 1`] = `
<div>
  <svg>
    <g
      class="cursor"
      transform="translate(324 0)"
    >
      <path
        d="M572.2,346.6c-16,0-29,13-29,29s13,29,29,29c16,0,29-13,29-29S588.1,346.6,572.2,346.6z"
        style="fill: #FDFDFD;"
      />
      <path
        d="M572.2,336.6c-21.5,0-39,17.5-39,39s17.5,39,39,39c21.5,0,39-17.5,39-39S593.7,336.6,572.2,336.6z
		 M543.2,375.6c0-16,13-29,29-29c16,0,29,13,29,29s-13,29-29,29C556.2,404.6,543.2,391.6,543.2,375.6z"
        style="fill: #273259;"
      />
    </g>
  </svg>
</div>
`;

exports[`<PlanetScoreCursor> (E) renders the PlanetScoreCursor 1`] = `
<div>
  <svg>
    <g
      class="cursor"
      transform="translate(432 0)"
    >
      <path
        d="M572.2,346.6c-16,0-29,13-29,29s13,29,29,29c16,0,29-13,29-29S588.1,346.6,572.2,346.6z"
        style="fill: #FDFDFD;"
      />
      <path
        d="M572.2,336.6c-21.5,0-39,17.5-39,39s17.5,39,39,39c21.5,0,39-17.5,39-39S593.7,336.6,572.2,336.6z
		 M543.2,375.6c0-16,13-29,29-29c16,0,29,13,29,29s-13,29-29,29C556.2,404.6,543.2,391.6,543.2,375.6z"
        style="fill: #273259;"
      />
    </g>
  </svg>
</div>
`;

exports[`<PlanetScoreCursor> should set translateY value 1`] = `
<div>
  <svg>
    <g
      class="cursor"
      transform="translate(0 200)"
    >
      <path
        d="M572.2,346.6c-16,0-29,13-29,29s13,29,29,29c16,0,29-13,29-29S588.1,346.6,572.2,346.6z"
        style="fill: #FDFDFD;"
      />
      <path
        d="M572.2,336.6c-21.5,0-39,17.5-39,39s17.5,39,39,39c21.5,0,39-17.5,39-39S593.7,336.6,572.2,336.6z
		 M543.2,375.6c0-16,13-29,29-29c16,0,29,13,29,29s-13,29-29,29C556.2,404.6,543.2,391.6,543.2,375.6z"
        style="fill: #273259;"
      />
    </g>
  </svg>
</div>
`;
